robottelo/api/utils.py

```diff
@@ -891,3 +891,24 @@ def attach_custom_product_subscription(prod_name=None, host_name=None):
         query={'search': 'name={0}'.format(prod_name)})[0]
     entities.HostSubscription(host=host.id).add_subscriptions(
         data={'subscriptions': [{'id': product_subscription.id, 'quantity': 1}]})
+
+
+def update_provisioning_template(name=None, old=None, new=None, org_id=None):
+    """ Update provisioning template content
+
+    :param str name: template provisioning name
+    :param str old: current content
+    :param str new: replace content
+    :param int org_id: organization ID
+
+    :return boolean: True/False
+    """
+    temp = entities.ProvisioningTemplate().search(
+        query={
+            'per_page': 1000,
+            'search': 'name="{}"'.format(name),
+            'organization_id': org_id
+        })[0].read()
+    temp.template = temp.template.replace(old, new)
+    update = temp.update(['template'])
+    return new in update.template
```

Comment:
If we pass values  update_provisioning_template('foo', 'too') having template as template = 'testing too', In this case the method will return True. 

Do you think we need to add one more layer of validation to see if old value truly exist in the template? As we are returning the update based on new value exist in the template.

Reply:
@omkarkhatavkar That is not added to avoid the condition where you have similer as below and condition will be shown always a false `True` hence to avoid that only good to validate the new:

```
i.e.  in below new replaced with #new :

                                                                                                                                     
In [4]: test = 'this is test on <#new>'                                                                                                                                      

In [5]: 'new' in test                                                                                                                                                        
Out[5]: True

In [6]: '#new' in test                                                                                                                                                       
Out[6]: True
```

Reply:
@vijay8451 My point was before updating try to check if the old value really exists in the template. I was not saying you to check the new value in the template. If we don't have the old value nothing will replace and we will only update the same thing. 

Whatever you mentioned above has the old value, which means the replacement will work and get updated to the new value. My thought process was to see if an old value exists in the template to ensure that replacement was correct and update calls make sense with checking new value. 

I'm ok if you think that we don't want that validation or assuming this condition will never occur.

Reply:
Done :)